.github/workflows/stale-prs.yml

```diff
@@ -0,0 +1,113 @@
+                try{
+                  const reviews = await github.rest.pulls.listReviews({
+                    owner: 'rancher',
+                    repo: 'shepherd',
+                    pull_number: pr.number,
+                    per_page: 100,
+                  });
+                  
+                  if ( reviews.data.length == 0 && createdDate < fiveDaysAgo ) {
+                    noReviewsReport += `${pr.title} #${pr.number} ${pr.html_url} ${pr.user.login}\n`;
+                  } 
+                  else {
+                      for (const review of reviews.data) {
+                        const reviewDate = new Date(review.submitted_at);
+                        if(!lastReviewDate || reviewDate > lastReviewDate) {
+                          lastReviewDate = reviewDate
+                          lastReviewStateTwo = lastReviewStateOne
+                          lastReviewStateOne = review.state
+                          if ( review.state === "APPROVED"){
+                            numberOfApprovedState+=1
+                          }
+                      }
+                    }
+                  }
+                } catch( error ){
+                  hasErrors += `${pr.title} (#${pr.number}) ${pr.html_url} - Error: ${error} \n`;
+                }
+                if ( lastReviewDate && (lastReviewDate < threeDaysAgo && updatedDate <= lastReviewDate) && statesCompare.includes(lastReviewStateOne) && statesCompare.includes(lastReviewStateTwo) ) {
+                  needsChanges += `${pr.title} (#${pr.number}) ${pr.html_url} ${pr.user.login} \n`;
+                  
+                } else if ( lastReviewDate && (updatedDate > lastReviewDate) && (updatedDate < threeDaysAgo) ){
+                  needsReviews += `${pr.title} (#${pr.number}) ${pr.html_url} ${pr.user.login} \n`;
+
+                }  else if ( lastReviewDate && numberOfApprovedState >=2 && (lastReviewDate < threeDaysAgo) && !statesCompare.includes(lastReviewStateOne) ) {
+                  needsMerge += `${pr.title} (#${pr.number}) ${pr.html_url} ${pr.user.login} \n`;
+                }
+              }
+              if ( noReviewsReport ) {
+                report += `PRs with no reviews yet: \n` + noReviewsReport +`\n\n`
+              }
+              if ( needsReviews ) {
+                report += `PRs that need reviews: \n`+ needsReviews +`\n\n` 
+              }
+              if ( needsChanges ) {
+                report += `PRs with changes requested: \n`+ needsChanges +`\n\n`
+              }
+              if ( needsMerge ) {
+                report += `PRs that need to be merged: \n` + needsMerge
+              }
+              if ( hasErrors ) {
+                report += `PRs that have errors when fetching: \n` + hasErrors
+              }
+              return report;
+
+            } catch( error ){
+              report += `There is an error getting PR Status report - Error: ${error} \n`;
```

Comment:
since this is adding to the original report, will like 110 ever hit?

Reply:
maybe? I don't know. I couldn't test that in this repo because there are not many open PRs.